src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -264,22 +264,27 @@ async def _finish_handler_with_unary_response(RPCState rpc_state,
     rpc_state.raise_for_termination()
 
     # Serializes the response message
-    cdef bytes response_raw = serialize(
-        response_serializer,
-        response_message,
-    )
+    cdef bytes response_raw
+    cdef SendMessageOperation send_message_op
```

Comment:
I guess that we could avoid creating this temporary variable and doing two checks for the status code by moving the `SendMessageOperation` construction just right after `SendStatusFromServerOperation`.

am I right? or am I missing something?

Reply:
Done.

I wasn't sure the consequence of skipping the `SendMessageOperation`, because I was seeing a cancellation log. But after correcting the cancellation condition, skipping the `SendMessageOperation` is not the culprit.